pandas/tests/groupby/test_indexing.py

```diff
@@ -314,3 +314,18 @@ def test_groupby_duplicated_columns(func):
         [[1.5, 3.0, 1.5]], columns=["A", "B", "A"], index=pd.Index(["G"], name="C")
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_dataframe_groupby_incorrect_with_multiindex_make_df_from_data():
+    # GH#32492
+    df = pd.DataFrame(
+        data={
+            "A": ["a1", "a2", None],
+            "B": ["b1", "b2", "b1"],
+            "val": [1, 2, 3],
+        }
+    )
+    grps = df.groupby(by=["A", "B"])
+
+    with pytest.raises(KeyError):
```

Comment:
Can you add a match statement? I think pre-commit is complaining about this too

Reply:
Please let the reviewer mark conversations as resolved

I mean: 

``pytest.raises(KeyError, match="String to match"):``